programs/ziti-edge-tunnel/instance.c

```diff
@@ -286,6 +287,42 @@ tunnel_port_range *getTunnelPortRange(ziti_port_range *zpr){
     return tpr;
 }
 
+static void setTunnelAllowedSourceAddress(tunnel_service *tnl_svc, ziti_service *service) {
+    const char* cfg_json = ziti_service_get_raw_config(service, CFG_HOST_V1);
+    tunnel_address_array allowed_src_addr_arr = NULL;
+    if (cfg_json != NULL && strlen(cfg_json) > 0) {
+        ZITI_LOG(TRACE, "host.v1: %s", cfg_json);
+        ziti_host_cfg_v1 cfg_v1 = {0};
+        parse_ziti_host_cfg_v1(&cfg_v1, cfg_json, strlen(cfg_json));
+        size_t n = 0;
+        int j = 0;
+        ziti_address_array allowed_src_addrs = cfg_v1.allowed_source_addresses;
+        for (int x = 0; allowed_src_addrs != NULL && allowed_src_addrs[x] != NULL; x++) {
+            n++;
+        }
+        allowed_src_addr_arr = calloc(n + 1, sizeof(tunnel_address *));
+        for (int i = 0; i < n; i++) {
+            if (allowed_src_addrs[i]->type != ziti_address_cidr) {
```

Comment:
minor: if this test happened in the loop above where `n` is calculated, you could allocate precisely enough memory for `allowed_src_addr_arr`.

Reply:
The reason I did not was because I would have to run the same test again in the second loop since I don't have a way to know which entries were discarded from the original array and did not want redundant code and assumed it should be rare that someone actually puts a hostname instead of a prefix since it would be in error.  If you prefer it I can add it.  let me know.